ipi/engine/motion/dynamics.py

```diff
@@ -361,12 +365,11 @@ def pconstraints(self):
             vcom = np.sum(p.reshape(-1, 3), axis=0) / Mnb
             beads.p -= (m3 * vcom).reshape(nb, -1)
 
-            self.ensemble.eens += np.sum(vcom**2) * 0.5 * Mnb  # COM kinetic energy
+            self.ensemble.eens += np.sum(vcom**2) * 0.5 * Mnb  # COM kinetic energy.
 
         if len(self.fixatoms_dof) > 0:
             m3 = dstrip(beads.m3)
             p = dstrip(beads.p)
-
             self.ensemble.eens += 0.5 * np.sum(
                 p[:, self.fixatoms_dof] ** 2 / m3[:, self.fixatoms_dof]
             )
```

Comment:
here we are setting the the momenta of the fix dof to zero, 
The changes in line 411ish mean that the momenta of those dof is not updated.

I might be missing something, but I don't understand why we need to do both things.
Do these changes fix the drift you observe?